Web_Development/Javascript/Topics/6.Variables/README.md

```diff
@@ -0,0 +1,273 @@
+    </script>
+</body>
+</html>
+```
+## Javascript Const Variable
+Variables declared with the const maintain constant values. const declarations share some similarities with let declarations.**const declarations are block scoped.**
+Like let declarations, const declarations can only be accessed within the block they were declared.
+## const cannot be updated or re-declared
+
+This means that the value of a variable declared with const remains the same within its scope. It cannot be updated or re-declared. So if we declare a variable with const
+
+## When to use JavaScript const?
+
+As a general rule, always declare a variable with const unless you know that the value will change.
+
+Use **const** when you declare:
+   1. A new Array
+   2. A new Object
+   3. A new Function
+   4. A new RegExp
+
+## Example 1: It explains that the const variable cannot be reassigned.
+```html
+<html>
+<head>
+    <title>JavaScript const variable cannot be reassigned example</title>
+</head>
+<body>
+    <script type="text/javascript">
+            const x = 12;
+            x = 13;
+            x += 1;
+    </script>
+</body>
+</html>
+```
+## Output of the above example
+
+![image](word-image-109.png)
+
+## Example 2: It explains the const variable which contains the Block Scope.
+```js
+<html>
+<head>
+    <title>JavaScript const variable which contains the Block Scope example</title>
+</head>
+<body>
+    <script type="text/javascript">
+        const x = 22;
+        {
+            const x = 90;
+            console.log(x);
+
+            {
+                const x = 77;
+                console.log(x);
+            }
+            {
+                const x = 45;
+                console.log(x);
+            }
+        }
+        console.log(x);
+    </script>    
+</body>
+</html>
+```
+## Output of the above example
+
+![image](word-image-110.png)
+
+## Types of Javascript Variables
+
+A JavaScript variable is simply a name of storage location. There are two types of variables in JavaScript : **local variable and global variable**.
+
+## Javascript Local Variable
+A JavaScript local variable is declared inside block or function. It is accessible within the function or block only.
+
+__Source Code:__
+```html
+  <html>
+    <body>
+      <script>  
+       function abc(){  
+          var x=10;//local variable  
+        }  
+     </script>  
+    </body>
+  </html>
+```
+## JavaScript global variable
+
+A JavaScript global variable is accessible from any function. A variable i.e. declared outside the function or declared with window object is known as global variable.
+
+__Source Code:__
+
+```html
+<html>
+ <body>
+   <script>  
+     var data=200;//gloabal variable  
+     function a(){  
+      document.writeln(data);  
+     }  
+     function b(){  
+       document.writeln(data);  
+      }  
+      a();//calling JavaScript function
+      b();
+   </script>  
+ </body>
+</html>
+```
+## Output of the above example
+```
+200 200
+```
```

Comment:
`document.writeIn()` prints in the body and not in console
So if you want to use `document.writeIn()` then attach a screenshot of your webpage with `200 200`

Reply:
ok sir